### ISSUE TITLE
bug(cdk-overlay-dark-backdrop): Is solid black in Windows high contrast dark mode

### ISSUE THREAD

**Issue by haywoodsloan** [angular/components#21890]
#### Reproduction

Steps to reproduce:
1. On a Windows machine enable high contrast mode with the black theme
2. Using Firefox navigate to: https://stackblitz.com/edit/angular-o7autm?file=src/app/./cdk-overlay-basic-example.html
3. Click the 'Open' button to show the overlay with backdrop enabled
4. See the overlay is solid black and obscures the rest of the page
 
 
#### Expected Behavior

The overlay does not obscure the rest of the page, such as the non-HC behavior: 
![image](https://user-images.githubusercontent.com/5940783/107827960-aba33f80-6d3c-11eb-8a97-2968c732748d.png)

#### Actual Behavior

The transparent overlay backdrop is converted to a solid black color and obscures everything else on the page:
![image](https://user-images.githubusercontent.com/5940783/107828103-02a91480-6d3d-11eb-93a5-2fb4828802e3.png)

#### Environment

- Angular: 80f4ff33381d37e72764d2ea3b7060e5952232e9
- CDK/Material: 70ff528785cb9abe9f102f19b1ea87caef56a295
- Browser(s): Firefox
- Operating System (e.g. Windows, macOS, Ubuntu): Windows
